### PR TITLE
Gitleaks issues

### PR DESCRIPTION
**Pull request by Bisonburger**
This PR fixes gitleaks

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

## Changes
- added support for 192.* IP 4 addresses
- excluded /frontend/node_modules and /frontend/dist
- excluded local vault changes
- reset react build

### Testing
- [ ] Tested locally?
- [ ] Ran `quickTest all`?
- [ ] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

### Specific Security-related subjects a reviewer should pay specific attention to

- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
*List GitHub issues this PR fixes*
- #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

### CODE REVIEW

File: .environment/gitleaks/gitleaks-config.toml [hunk 1]
```diff
@@ -16,12 +16,16 @@ title = "PRIME ReportStream Gitleaks Configuration"
         '.terraform/providers/',
         'frontend-react/build/',
         'frontend-react/node_modules/',
+        'frontend/node_modules/',
+        'frontend/dist',
+        'frontend/src/assets',
```
Comment (td-usds) [inline .environment/gitleaks/gitleaks-config.toml:21]: This I'm not totally comfortable with; can this instead be done by file type? What if we put JS in there or anything else...? We'd want the JS to be scanned, but -for instance- not fonts or JPEGs.

File: .environment/gitleaks/gitleaks-config.toml [hunk 2]
```diff
@@ -244,6 +248,7 @@ title = "PRIME ReportStream Gitleaks Configuration"
             '([0-1])\.0\.0\.0',
             '\d+\.\d+\.\d+\.\d+\.',                     # Fix for bug in regex (allows trailing dots)
             '1\.1\.1\.1',                               # Public DNS
+            '192\.(\d+\.){2}\d+',                        # Cheapo 10.0.0.0/8 range
```
Comment (td-usds) [inline .environment/gitleaks/gitleaks-config.toml:251]: 1. Comment is out of sync
2. check out 3f52c6483 (a commit by @ronaldheft-usds) which should address this; do a merge from `master` -> `gitleaks-issues` and eliminate this/your change to see if it fixes what you were seeing.

File: .github/workflows/build_frontend.yaml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
           list-files: csv
           filters: |
             frontend:
-              - 'frontend-react/**'
+              - 'frontend/**'
```
Comment (td-usds) [inline .github/workflows/build_frontend.yaml:27]: Is this intended to be part of this PR?

**Review by td-usds — CHANGES_REQUESTED** [review]

**Comment by td-usds** [comment]
I still see a handful of failures being highlighted when I run this on your branch (i.e. the gitleaks-config.toml does not cover all the failures that need to be fixed). Most of them are in .tsx files containing html along the lines of `<something>key="somethinmg"`

File: .environment/gitleaks/gitleaks-config.toml (hunk 1 above)
Comment (mry-usds) [reply .environment/gitleaks/gitleaks-config.toml:21]: Frontend is going away by end of this sprint -- lets not make this too complex

File: .environment/gitleaks/gitleaks-config.toml (hunk 2 above)
Comment (mry-usds) [reply .environment/gitleaks/gitleaks-config.toml:251]: Done

File: .github/workflows/build_frontend.yaml (hunk 3 above)
Comment (mry-usds) [reply .github/workflows/build_frontend.yaml:27]: Yes - we need to revert the changes to deploy frontend-react

Comment (td-usds) [reply .github/workflows/build_frontend.yaml:27]: we need to revert this _full stop_ or _we need to revert this as part of this PR_ ?

Comment (Bisonburger) [reply .github/workflows/build_frontend.yaml:27]: The line is incorrect and will not work properly...this needs to be reverted

**Comment by Bisonburger** [comment]
The additional key="value" is tied up in another PR which was merged just now

File: .environment/gitleaks/gitleaks-config.toml (hunk 2 above)
Comment (td-usds) [reply .environment/gitleaks/gitleaks-config.toml:251]: See line 256, we have coverage for this (which is why I wanted that merged in :) ); this line can go now

File: .github/workflows/build_frontend.yaml (hunk 3 above)
Comment (td-usds) [reply .github/workflows/build_frontend.yaml:27]: Discussed in person

**Review by td-usds — CHANGES_REQUESTED** [review]

File: .environment/gitleaks/gitleaks-config.toml (hunk 1 above)
Comment (td-usds) [reply .environment/gitleaks/gitleaks-config.toml:21]: Discussed in person, @mry-usds is correct

**Pull request closed without merging**